impeller/aiks/aiks_unittests.cc

```diff
@@ -3581,5 +3581,51 @@ TEST_P(AiksTest, VerticesGeometryUVPositionData) {
   ASSERT_TRUE(OpenPlaygroundHere(canvas.EndRecordingAsPicture()));
 }
 
+TEST_P(AiksTest, ClearBlendWithBlur) {
+  Canvas canvas;
+  Paint white;
+  white.color = Color::White();
+  canvas.DrawRect(Rect::MakeXYWH(0, 0, 600.0, 600.0), white);
+
+  Paint clear;
+  clear.blend_mode = BlendMode::kClear;
+  clear.mask_blur_descriptor = Paint::MaskBlurDescriptor{
+      .style = FilterContents::BlurStyle::kNormal,
+      .sigma = Sigma(20),
+  };
+
+  canvas.DrawCircle(Point::MakeXY(300.0, 300.0), 200.0, clear);
+
+  ASSERT_TRUE(OpenPlaygroundHere(canvas.EndRecordingAsPicture()));
+}
+
+TEST_P(AiksTest, ClearBlend) {
+  Canvas canvas;
+  Paint white;
+  white.color = Color::White();
+  canvas.DrawRect(Rect::MakeXYWH(0, 0, 600.0, 600.0), white);
+
+  Paint clear;
+  clear.blend_mode = BlendMode::kClear;
+
+  canvas.DrawCircle(Point::MakeXY(300.0, 300.0), 200.0, clear);
+
+  ASSERT_TRUE(OpenPlaygroundHere(canvas.EndRecordingAsPicture()));
+}
+
+TEST_P(AiksTest, ClearBlendImage) {
```

Comment:
Ah, there's actually already a golden for this, but it's kind of sneaky because the correct behavior is for it to render nothing, unlike all the other blend variants. :) `AiksTest.BlendModeClear`

It looks like both the existing golden and this new golden aren't quite rendering correctly with this change: https://api.flutter.dev/flutter/dart-ui/BlendMode.html#clear

Reply:
Oh wow, the golden results are surprising, there is a lot to unpack:
- The new clear test `impeller_Play_AiksTest_ClearBlend_*` renders nothing correctly
- `impeller_Play_AiksTest_ClearBlendWithBlur_Vulkan` gives me different results locally with moltenvk
- The existing clear test `impeller_Play_AiksTest_BlendModeClear_Metal` rendered nothing incorrectly, but this PR seems to have broken that (I would have expected my change to only effect rrect blur). This is also surprising considering `impeller_Play_AiksTest_ClearBlend_*`
- `impeller_Play_AiksTest_ClearBlendImage_Vulkan` is also giving me different results locally

I'll see if I can get this sorted out.

## local impeller_Play_AiksTest_ClearBlendImage_Vulkan

<img width="622" alt="Screenshot 2023-09-26 at 8 50 25 AM" src="https://github.com/flutter/engine/assets/30870216/98a637c6-8f35-43a3-b8b5-25d57b9becdb">

## local impeller_Play_AiksTest_ClearBlendWithBlur_Vulkan

<img width="578" alt="Screenshot 2023-09-26 at 8 50 09 AM" src="https://github.com/flutter/engine/assets/30870216/c3027675-3fc7-4e54-847d-b9252afc0b8c">

Reply:
I had been working with the vulkan tests, just verified that locally the metal tests give different/correct results.  I wonder if skiagold got messed up.  It is reporting https://github.com/flutter/engine/commit/98cec738eb342d39ad9a99fd7441bd9cec596f96 though.

Reply:
Filed an issue about the discrepancy between skiagold and local results: https://github.com/flutter/flutter/issues/135507

I thought maybe it was a moltenvk thing, but metal and vulkan are giving the same correct results locally which differ from skiagolds results.

Reply:
I removed `ClearBlendImage` since it's a duplicate.  We'll also see if pushing a new commit flushes out the skiagold issue.

Reply:
Huh, weird. Yeah I couldn't spot anything that should cause the existing golden to fail yesterday. Thanks for digging in!